src/js/server/lib/datasource.js

```diff
@@ -0,0 +1,160 @@
+// A rewritten version of `kettle.dataSource.CouchDB.read` that allows rewriting of the response using the
+// transformation rules found in `options.rules.read`.  We cannot directly extend the existing couch grade in Kettle
+// because it has a non-namespaced listener for `onRead`, which we cannot replace. The `gradeNames` and `listener` block
```

Comment:
A listener to an event will displace another one on the same event with the same namespace. If you write a definition just like this one in a derived grade or in an options distribution, it will override this one. This particular mechanism isn't new, it has been implemented like this from the beginnings of the framework - although I remember that you and I established in a chat last year that it was, in the end, based on a "creative misunderstanding" of the jQuery system which it was believed to be modelled on. It turns out that jQuery namespaces don't behave like this at all - they are used at the "firing" end to select just a subset of the available listeners for a particular firing of an event. All of our listeners that are registered always fire all the time.